sky/sdk/example/game/main.dart

```diff
@@ -90,13 +91,77 @@ class GameDemoApp extends App {
 
   Widget _buildMainScene(navigator, route) {
     return new Center(
-      child: new RaisedButton(
-        child: new Text("Play"),
+      child: new TextureButton(
         onPressed: () {
           _game = new GameDemoWorld(_app, navigator, _loader, _spriteSheet, _spriteSheetUI);
           navigator.pushNamed('/game');
-        }
+        },
+        texture: _spriteSheetUI["btn_play_up.png"],
+        textureDown: _spriteSheetUI["btn_play_down.png"],
+        width: 128.0,
+        height: 128.0
       )
     );
   }
 }
+
+class TextureButton extends ButtonBase {
+  TextureButton({
+    Key key,
+    this.onPressed,
+    this.texture,
+    this.textureDown,
+    this.width: 128.0,
+    this.height: 128.0
+  }) : super(key: key);
+
+  Function onPressed;
+  Texture texture;
+  Texture textureDown;
+  double width;
+  double height;
+  bool _pressed = false;
+
+  int _token = 0;
+
+  void syncFields(TextureButton source) {
+    onPressed = source.onPressed;
+    texture = source.texture;
+    textureDown = source.textureDown;
+    super.syncFields(source);
```

Comment:
Also since you support changing the texture on the fly here you should probably change the token here too.